qiskit_experiments/framework/experiment_data.py

```diff
@@ -913,20 +930,24 @@ def _run_analysis_callback(
             LOG.warning(error_msg)
             return callback_id, False
 
-    def _add_result_data(self, result: Result) -> None:
+    def _add_result_data(self, result: Result, job_id: Optional[str] = None) -> None:
         """Add data from a Result object
 
         Args:
             result: Result object containing data to be added.
+            job_id: The id of the job the result came from. If `None`, the
+            job id in `result` is used.
```

Comment:
1. It's important to understand what's the job id that's stored in `Result`. Could it be a bug?
2. Depending on the answer of (1), decide whether to keep `job_id` optional like here. If the job id in `Result` is meaningless then why keep this option.

Reply:
We might want to avoid using the id in `Result`. It used to be the same id that is used to obtain the job from which the result came, but something seems to have changed, server side (this is bug 1 fixed by this PR). However, when `_add_result_data` is used from `add_data` (instead of `_add_job_data` called by the job handler) we only get the `Result` object itself, so if we're not using the data in `Result` we'll have to treat it as `None`.

Reply:
I think the effort should be made to understand what has changed and why

Reply:
The exact change is like this: when retrieving a result of a job from the server using, e.g. the job_id `cgvq30fm7dri9b4n4gm0`, you will receive a dictionary of the result data, which does not have this id `cgvq30fm7dri9b4n4gm0` as part of its data. Instead, there will be a field `job_id` with a different value (and value format), e.g. `44beed55-c580-4821-884a-a1d463eabc19`.

We still need the original job_id `cgvq30fm7dri9b4n4gm0` in order to access job data; I currently see no use to the new job_id; it cannot be used to retrieve data. We may want to ignore it altogether for now.